### DISCUSSION TITLE
Retry Configuration

### DISCUSSION THREAD

**Discussion by santoshj88** [operator-framework/java-operator-sdk#1278] [Q&A] [answered]
We are using io.javaoperatorsdk version 2.1.4 to create kubernates operator How can we configure retry configuration. I went through the doc but didnot understand the way how to configure. could some one help on this ?

I have code like below

@ControllerConfiguration
public class TestReconciler implements Reconciler<TestCustomResource>,
    ErrorStatusHandler<TestCustomResource>,
    EventSourceInitializer<TestCustomResource> {

....
.....
}

KubernetesClient client = new DefaultKubernetesClient(config);
Operator operator = new Operator(client,
        TestOperatorConfigurationServiceOverrider.getService());
operator.register(new TestReconciler(client));

**Comment by metacosm** [comment]
You'd do something like:

```java
operator.register(new ApplicationsReconciler(client), configOverrider -> configOverrider
            .withRetry(new RetryConfiguration() {
              @Override
              public int getMaxAttempts() {
                return 10;
              }
            }));
```

In this example, you'd be overriding the maximum number of retry attempts to 10 instead of the default value.

**Reply by santoshj88** [reply]
withRetry method is not part of 2.1.4 which is approved for us . could you please share solution for 2.1.4 version ?

**Comment by csviri** [comment, marked as answer]
The retry instance can be used directly, also:

```java
operator.register(schemaReconciler,
        configOverrider -> configOverrider.withRetry(new GenericRetry().setMaxAttempts(10)));
```

**Reply by santoshj88** [reply]
withRetry method is not part of 2.1.4 version which is approved for us . could you please share solution for 2.1.4 version ?

**Reply by csviri** [reply]
in 2.1.4 like this:

```java
KubernetesClient client = new DefaultKubernetesClient();
    Operator operator = new Operator(client, DefaultConfigurationService.instance());
    
    TomcatReconciler reconciler = new TomcatReconciler(client);
    operator.register(new TomcatReconciler(client),ControllerConfigurationOverrider
            .override(new AnnotationConfiguration<>(reconciler)).withRetry(new GenericRetry().setMaxAttempts(10))
            .build());
    
    operator.installShutdownHook();
    operator.start();
```

(pls consider upgrading to v3.0.x )
